### PR TITLE
feat: Reduce install size

### PR DESCRIPTION
**Pull request by eps1lon**
BREAKING CHANGE:
1. Needs `Map` and `Set` polyfills if these built-ins are not available
1. `aria-query/lib/*` no longer exports `Map` but objects.
    Use named imports instead.
    ```diff
    -import roles from 'aria-query/lib/roleElementMap'
    +import { roles } from 'aria-query'
    ```

Drops usage of babel/runtime

Closes https://github.com/A11yance/aria-query/pull/162
Closes https://github.com/A11yance/aria-query/pull/166
Closes https://github.com/A11yance/aria-query/issues/158

Test plan
- [x] `eslint-plugin-jsx-a11y` (failing tests are related to ARIA 1.2 update)
- [x] `@testing-library/dom`

/cc @TimvdLippe

### CODE REVIEW

**Comment by TimvdLippe** [comment]
I can confirm this would resolve the various PRs/issues I filed. Thank you so much for putting the time in this! Very excited for seeing this landing 🎉

**Comment by jessebeach** [comment]
@eps1lon this will require updates to eslint-plugin-jsx-a11y, right?

**Comment by eps1lon** [comment]
> @eps1lon this will require updates to eslint-plugin-jsx-a11y, right?

It shouldn't as far as I can tell. I did try it already but `aria-query` hasn't been released in a while so the failures I did see looked like they're related to all the ARIA 1.2 updates. 

We may want to release a major of `aria-query` from the latest `main` to propagate all the ARIA 1.2 changes and then check again.

File: .babelrc [hunk 1]
```diff
@@ -8,11 +8,6 @@
     "@babel/preset-flow"
   ],
   "plugins": [
-    [
-      "@babel/plugin-transform-runtime", {
```
Comment (gatsbimantico) [inline .babelrc:12]: You dropped the usage of babel runtime, but I think you still left it on the package.json

**Comment by jessebeach** [comment]
We'll do a release next week. I've got time to work on this project then.

**Comment by eps1lon** [comment]
> We'll do a release next week. I've got time to work on this project then.

With this PR or without? I'd prefer with this PR. We can always release fixes if there's something missing.

**Comment by nickmccurdy** [comment]
I think this should raise the required Node version in `engines`.

**Comment by eps1lon** [comment]
> I think this should raise the required Node version in `engines`.

Could you elaborate why you think this is the case?

**Comment by nickmccurdy** [comment]
Is the Babel output still compatible with Node 6 with that plugin removed?

**Comment by eps1lon** [comment]
> Is the Babel output still compatible with Node 6 with that plugin removed?

As far as I can tell, no. Why did you raise that concern though?

**Comment by nickmccurdy** [comment]
Because `engines` is currently set to `>=6.0` in `package.json`.

**Comment by eps1lon** [comment]
>  While I still think this is a good idea, it seems like we should signal that Node 6 or later isn't compatible by updating engines.

It's still unclear to me what is incompatible. You already made that claim but didn't show how compatibility changed. Could you please be more specific what broke node 6 compat?

**Comment by ljharb** [comment]
You just said the babel output is no longer compatible with node 6 - is that correct?

**Comment by eps1lon** [comment]
> You just said the babel output is no longer compatible with node 6 - is that correct?

Sorry, typo in https://github.com/A11yance/aria-query/pull/180#issuecomment-884704846

The breaking changes are only the ones listed in the PR description.

**Comment by ljharb** [comment]
In that case, is the current engines.node of ">= 6" perhaps overzealous? Map and Set are available in node >= 0.12.

**Comment by eps1lon** [comment]
> In that case, is the current engines.node of ">= 6" perhaps overzealous? Map and Set are available in node >= 0.12.

Yeah, but this is somewhat of a different issue which we would need to test.

**Comment by jessebeach** [comment]
#250 should address the need here.

**Pull request closed without merging**